src/configure/templateInputHelper/InputControlProvider.ts

```diff
@@ -81,6 +86,42 @@ export class InputControlProvider {
         });
     }
 
+    private _initializeDynamicValidations() {
+        this._pipelineTemplate.inputs.forEach((inputDes) => {
+
+            var inputControl = this._inputControlsMap.get(inputDes.id);
+
+            if (!!inputControl && !!inputDes.dynamicValidations && inputDes.dynamicValidations.length > 0) {
+                var dataSourceToInputsMap = new Map<DataSource, InputControl[]>();
+                inputDes.dynamicValidations.forEach((validation: InputDynamicValidation) => {
+                    var validationDataSource = DataSourceUtility.getDataSourceById(this._pipelineTemplate.dataSources, validation.dataSourceId);
+                    if (validationDataSource) {
+                        dataSourceToInputsMap.set(validationDataSource, []);
+
+                        let dynamicValidationRequiredInputIds = DataSourceUtility.getDependentInputIdList(validationDataSource.endpointUrlStem);
+                        dynamicValidationRequiredInputIds = dynamicValidationRequiredInputIds.concat(DataSourceUtility.getDependentInputIdList(validationDataSource.requestBody));
+                        dynamicValidationRequiredInputIds = dynamicValidationRequiredInputIds.concat(DataSourceUtility.getDependentInputIdList(validationDataSource.resultTemplate));
+                        dynamicValidationRequiredInputIds = dynamicValidationRequiredInputIds.concat(DataSourceUtility.getDependentInputIdList(validationDataSource.resultSelector));
+
+                        dynamicValidationRequiredInputIds = Array.from(new Set(dynamicValidationRequiredInputIds));
+                        dynamicValidationRequiredInputIds.forEach((dynamicValidationRequiredInputId) => {
+                            var dependentInput = this._inputControlsMap.get(dynamicValidationRequiredInputId);
+                            if (dependentInput && dynamicValidationRequiredInputId !== inputDes.id) {
+                                dataSourceToInputsMap.get(validationDataSource).push(dependentInput);
+                            } else {
+                                //.logError("Dependent input {0} specified for input {1} is not present in pipeline template {2}".format(dynamicValidationRequiredInputId, inputDes.id, this._pipelineTemplate.id));
+                            }
+                        });
+                    } else {
+                        //PipelineTemplateErrorLogger.logError("validation data source {0} specified for input {1} is not present in pipeline template {2}".format(validation.dataSourceId, inputDes.id, this._pipelineTemplate.id));
```

Comment:
it would be good to trace these errors, as they help in figuring out if there are some issues in our templates

Reply:
I don't see any trace statement here

Reply:
you looked at older commit, trace statement is there.